### PR TITLE
fix(server): Init tx time for all multi/lua transactions

### LINKED ISSUE #2555: EVAL "return redis.call('TIME')" 0 returning zeros
**Describe the bug**
Executing `EVAL "return redis.call('TIME')" 0`

Returns:

1) (integer) "0"
2) (integer) "0"

**To Reproduce**
Execute the above command using the following approaches:
1. From the redis-cli inside the dragonfly docker image
2. StackExchange.Redis via lua script
3. RedisInsight workbench

All return:

1) (integer) "0"
2) (integer) "0"

Running TIME from redis-cli in docker image or Redis Insight correctly returns the system time values.

1) (integer) "1707230005"
2) (integer) "748298"

**Expected behavior**
The system time values should be returned using an EVAL script.

Docker image: sha256:e89f47ff91945e5f5661d82f48f302c0bd4acdd34630a1a754c8f862a1f5d0f5

### PR DESCRIPTION
**Pull request by chakaz**
Fixes #2555

### CODE REVIEW

**Review by kostasrim — DISMISSED** [review]

File: src/server/generic_family.cc [hunk 1]
```diff
@@ -1270,10 +1270,13 @@ void GenericFamily::Type(CmdArgList args, ConnectionContext* cntx) {
 }
 
 void GenericFamily::Time(CmdArgList args, ConnectionContext* cntx) {
-  uint64_t now_usec;
+  uint64_t now_usec = 0;
   if (cntx->transaction) {
+    // This will be 0 if run under an unscheduled transaction
```
Comment (romange) [inline src/server/generic_family.cc:1275]: how come we run under unscheduled transaction?

File: src/server/main_service.cc [hunk 2]
```diff
@@ -1710,8 +1710,10 @@ optional<bool> StartMultiEval(DbIndex dbid, CmdArgList keys, ScriptMgr::ScriptPa
     return false;
   }
 
-  if (keys.empty() && script_mode == Transaction::LOCK_AHEAD)
+  if (keys.empty() && script_mode == Transaction::LOCK_AHEAD) {
+    trans->InitTxTime();
```
Comment (adiholden) [inline src/server/main_service.cc:1714]: where do we call InitTxTime for NON_ATOMIC mode and keys is empty?

File: src/server/main_service.cc [hunk 3]
```diff
@@ -2040,7 +2042,9 @@ void Service::Exec(CmdArgList args, ConnectionContext* cntx) {
         "Dragonfly does not allow execution of a server-side Lua in Multi transaction");
 
   bool scheduled = false;
-  if (*multi_mode != Transaction::NOT_DETERMINED) {
+  if (*multi_mode == Transaction::NOT_DETERMINED) {
+    cntx->transaction->InitTxTime();
```
Comment (adiholden) [inline src/server/main_service.cc:2046]: how about call this in transaction constructor and if we will do scheduling this will overide the value

Comment (chakaz) [reply src/server/main_service.cc]: That's a fantastic idea!

**Pull request merged**